### DISCUSSION TITLE
OpenAI o1 on GitHub (Limited Public Preview) Feedback Thread

### DISCUSSION THREAD

**Discussion by 2percentsilk** [community/community#139125] [Copilot News and Announcements]
Starting today, we’re opening a preview to give developers an opportunity to [test OpenAI o1-preview and o1-mini](https://github.blog/news-insights/product-news/try-out-openai-o1-in-github-copilot-and-models/), hosted on Azure, in both GitHub Copilot and Models. 

[Sign up to get access to use OpenAI o1](https://github.com/o1-waitlist-signup?utm_campaign=discussions) in GitHub Copilot Chat with Visual Studio Code and in the playground with [GitHub Models](https://gh.io/models).

Please leave all feedback and questions below! We can't wait to see what you build 🚀

**Comment by ggoonnzzaallo** [comment]
I read that you launched o1-preview on GitHub Copilot Chat, but you actually launched a waitlist?

**Reply by novium2** [reply]
The wording is a bit ambiguous, isn't it? It's an opportunity to sign up for early access. You can read more here: [changelog](https://github.blog/changelog/2024-09-19-sign-up-for-openai-o1-access-on-github/)

**Reply by ggoonnzzaallo** [reply]
Still a good thing that the Copilot team is putting resources into keeping up with the latest models. I wonder if there will be new features enabled by o1 other than Chat

**Comment by cockscomb** [comment]
I received an email “You now have access to OpenAI o1 models on GitHub!” However, the “Please share any feedback with us on _Discussions_” link in the email redirects to `example.com`. The redirect destination might be incorrect.

**Reply by cockscomb** [reply]
I have confirmed that it has been fixed.

**Reply by novium2** [reply]
Thanks so much!

**Comment by ik-labs** [comment]
Isn't the waitlist available for individual paying customers? i get an error message saying `You must select an enterprise or organization to join the waitlist`?

**Reply by 2percentsilk** [reply]
If you are a Copilot Individual user, you should be able to sign up for the waitlist for yourself. However, if you receive a Copilot Business or Enterprise license, you need to nominate the organization for access.

**Reply by aamir-allaq** [reply]
How do I nominate the organization for access? I literally just see 1 button to join the waitlist. 
<img width="909" alt="image" src="https://github.com/user-attachments/assets/239d4d26-5ae1-4753-8c27-f826e1d6b365">

**Reply by Jerry-hyl** [reply]
Same. How can individual users have access to it?

**Reply by Gosis34** [reply]
Potrzebuję pomocy jak rozdzielić system android od komputera

**Comment by fn5** [comment]
Hooray! Excited to try it out within Copilot.

**Reply by Azeem075** [reply]
Yes

**Comment by Moulick** [comment]
> During this beta, you'll be able to experiment with the o1 models via [GitHub Models](https://docs.github.com/en/github-models/prototyping-with-ai-models), as well as Copilot Chat in Visual Studio Code with the new model selector.

Does anyone have a screenshot of where this model selector in Chat is supposed to be?

**Reply by novium2** [reply]
Stay tuned! More updates on the way.

**Reply by amankrokx** [reply]
I was wondering the same as I couldn't find the selector for o1 even in copilot chat pre prelease with vs code insiders

**Reply by arojoal** [reply]
I have received the email confirming that I have been given access to o1, I have installed latest version of VS code Studio Insiders and pre-release version of copilot chat and I can't find the Model Selector. Where is it?
<img width="540" alt="Screenshot 2024-09-25 at 08 01 29" src="https://github.com/user-attachments/assets/12b6da8b-83de-4113-bc9d-1cac7da76689">
<img width="262" alt="Screenshot 2024-09-25 at 08 00 57" src="https://github.com/user-attachments/assets/283d2dd8-fa2c-48be-bcac-e4404ab2d5e0">
<img width="722" alt="Screenshot 2024-09-25 at 08 02 04" src="https://github.com/user-attachments/assets/817631f9-6071-4ee9-8f7f-9adc792e6a6c">

**Comment by tsoernes** [comment]
What are the usage limits for the o1 models?

**Reply by AESMatias** [reply]
Hello @tsoernes, you can see it here (https://github.com/marketplace/models/azure-openai/gpt-4o):
![image](https://github.com/user-attachments/assets/b2cd36fe-a356-4e29-95bb-572dfeb731e4)

The rate limits of the plans: https://docs.github.com/en/github-models/prototyping-with-ai-models#rate-limits

**Reply by ggoonnzzaallo** [reply]
Wow so we get 12 o1 requests per day? that's more limited than ChatGPT

**Comment by cyoungberg** [comment]
I got the email saying I have access now... but how do I select the model? I made sure I'm up-to-date on the extensions, and I enabled the pre-release version of co-pilot chat. But not seeing any options?

<img width="471" alt="Screen Shot 2024-09-20 at 2 27 41 PM" src="https://github.com/user-attachments/assets/7ae36d33-1cc0-49a3-b837-dacb638b5dee">
<img width="191" alt="Screen Shot 2024-09-20 at 2 25 13 PM" src="https://github.com/user-attachments/assets/fd7b560e-1962-4abb-bf10-6d2c55b5e0a1">
<img width="185" alt="Screen Shot 2024-09-20 at 2 24 53 PM" src="https://github.com/user-attachments/assets/89d222bf-2015-4f5e-ab90-ee6141014fce">

**Reply by Moulick** [reply]
See the answer to my thread above

**Reply by cyoungberg** [reply]
ahh, sounds like someone jumped the gun before everything was ready maybe?

**Reply by fn5** [reply]
Haven't got an email, but checking, are you using VS Code Insiders as well?
I just tried to find where it said this...and I did.

Edit: 
https://github.blog/changelog/2024-09-19-sign-up-for-openai-o1-access-on-github/
![image](https://github.com/user-attachments/assets/e1313949-e496-47f6-a974-6861c897474a)

**Reply by cyoungberg** [reply]
Oop, I wasn't. Tried that now tho, and still no joy

**Reply by ggoonnzzaallo** [reply]
I think the model selection dropdown might be featured flag and we don't all have access to it yet?

Using a VS Code extension [double.bot](https://double.bot/) to get access to o1-preview right now, it's nice because there's no rate limits

**Reply by arojoal** [reply]
I have received the email confirming that I have been given access to o1, I have installed latest version of VS code Studio Insiders and pre-release version of copilot chat and I can't find the Model Selector. Where is it?
<img width="262" alt="Screenshot 2024-09-25 at 08 00 57" src="https://github.com/user-attachments/assets/0c238f24-ff51-49b8-9a56-8cb776f18485">
<img width="540" alt="Screenshot 2024-09-25 at 08 01 29" src="https://github.com/user-attachments/assets/e22a1057-d28c-4cc2-a875-f9137d97dc14">
<img width="722" alt="Screenshot 2024-09-25 at 08 02 04" src="https://github.com/user-attachments/assets/7f94520b-e505-4a3d-a4c1-f0678a2484b0">

**Comment by Axmoney** [comment]
> Starting today, we’re opening a preview to give developers an opportunity to [test OpenAI o1-preview and o1-mini](https://github.blog/news-insights/product-news/try-out-openai-o1-in-github-copilot-and-models/), hosted on Azure, in both GitHub Copilot and Models. 
> 
> [Sign up to get access to use OpenAI o1](https://github.com/o1-waitlist-signup?utm_campaign=discussions) in GitHub Copilot Chat with Visual Studio Code and in the playground with [GitHub Models](https://gh.io/models).
> 
> Please leave all feedback and questions below! We can't wait to see what you build 🚀 

Hello there,thanks for sharing

**Comment by sirusbaladi** [comment]
got in the waiting list, would love to get access if possible!

**Comment by Vikranth3140** [comment]
thanks for this :)

**Comment by Anselmoo** [comment]
@2percentsilk should also provide feedback in this thread, or will it be a separate discussion?
thx for feedback

**Reply by novium2** [reply]
Feedback in this thread is very welcome. Let us know what you think.

**Reply by 2percentsilk** [reply]
Please feel free to provide feedback here :)

**Reply by Anselmoo** [reply]
What definitely improves is the length of the output provided and keeping comments, aka, _docstrings_, with it if you explicitly ask for them. It is still a little strange why both models are cropping docstrings in default mode.

**Comment by Anselmoo** [comment]
One thing that often shows up is that the `GPT-4` model copies the answers from `o1-preview`, while if the `GPT-4` is used first, the `o1-preview` still delivers solutions in a different pedagogical format.


####  `o1-preview`-mode

Explains the way of the proposed refactor strategy:

<img width="547" alt="image" src="https://github.com/user-attachments/assets/2a4685e1-62b3-4926-8425-09ef050cd101">


#### `gpt4`-mode without previous  `o1-preview`-run

<img width="533" alt="image" src="https://github.com/user-attachments/assets/49a91561-5a17-4fba-971f-286bfe5c0b3e">

and with the previous run of  `o1-preview`, the results are identical. 

P.S. @2percentsilk and @novium2, is there a preferred format for how content should be posted; I am happy to learn

**Comment by ghost** [comment]
Security issues

**Comment by LIHUA919** [comment]
that's cool

**Comment by lcx-0504** [comment]
Sometimes I have needs to read some document and diagram. Is it possible for GitHub Copilot to support image and documents, as well as ability of online search?

**Reply by ghost** [reply]
Any help will be greatly appreciated

**Reply by Anselmoo** [reply]
### Introduction 
The attached files of the current workspace seem to work for both `vscode` and `vscode-insiders`.  

Used reference: https://github.com/Anselmoo/spectrafit/blob/main/spectrafit/plugins/notebook.py


### Step 1: Open the file
![image](https://github.com/user-attachments/assets/cbfa7d5a-c635-4537-89f0-93b45270b54d)

### Step 2: Selecting the corresponding file
![image](https://github.com/user-attachments/assets/0ed3a3c3-ea6d-44b7-bbc8-b17df379f5b5)

### Step 3: Define the task
![image](https://github.com/user-attachments/assets/f2087873-6b3d-40d0-9787-eac3c092fd1e)


### Step 4: Enjoy

@lcx-0504 is that what you are looking for?

P.S. It also works just by adding links into the prompt, and it will detect it.

**Reply by lcx-0504** [reply]
I know that I can attach files to chat. However, it seems that it only works for code files and text files. When I attach images or pdfs to copilot, it totally cannot understand it, and it seems to read it as "text" but not a file. Sometimes my colleagues send pdf or image or docx as description of function needed and I really don't want to copy every text to input box. So, it would be better for copilot to support those files? 😂 @Anselmoo

**Reply by Anselmoo** [reply]
Sorry, I didn't understand at first, but now I get your point, and Copilot does too.

> [!NOTE]  
> _It appears that you are referring to an image file named image.png. However, as a text-based AI, I cannot view or interpret images directly. If you can describe the content of the image or provide any relevant text or context from the image, I would be happy to help you understand or resolve any issues related to it._

**Reply by lcx-0504** [reply]
Yeah. That's just what I want say. I do hope image & dock & PDF could be support by Copilot. That could be really useful😊

**Comment by ghost** [comment]
Problem I’m suffering is the devices with full access to my accounts ex MAC
OC and Android many more which are directly and indirectly connected to my
accounts gmails Apple ID social media accounts etc . Please help me with
that .

ZiaHidary.$$$

**Comment by yan2441** [comment]
i faced a problem with the preview extension that some times i need to reload the window of vs in order to be able to use copilot 
also some times i send him and prompt and it start to think and after a while it did not return any answer

**Comment by ohofherr** [comment]
I signed up for o1 in GitHub Copilot last week and hope that we will get access soon.

Is it somewhere in the GitHub Organization visible when the access has been granted? I'm a little bit worried that the confirmation email could be filtered as spam.

**Comment by nikzan** [comment]
I received a letter that I was granted access to participate in the limited beta, but I still can't use the o1 model, they say I still need to sign up for the waiting list. Maybe someone has encountered something similar?

<img width="587" alt="Screenshot 2024-10-12 at 02 11 18" src="https://github.com/user-attachments/assets/64c7964f-44d7-47e1-9eda-84bdab6a83b5">
<img width="1335" alt="Screenshot 2024-10-12 at 02 11 53" src="https://github.com/user-attachments/assets/9cd4c6b5-dbb9-4674-98f1-6e8ad2c71c6d">

**Reply by thispaul** [reply]
The GitHub Models feature and o1 model access are two separate beta waitlists, sorry for the confusion. 
Please sign up for the o1 waitlist here: https://github.com/o1-waitlist-signup

**Reply by nikzan** [reply]
Thank you for the explanation!

**Comment by Anselmoo** [comment]
It would be cool if **OpenA1** is also available as an inline prompt for the `Jupyter` extension so `Jupyter` cells can be updated faster and, in my opinion, safer instead of jumping between cell and chat windows.

Otherwise, I can say that using **OpenA1** in combination with `Jupyter`  really helps a lot with post-structuring data cleaning and ensuring consistency by following the provided explanations.

**Comment by binaryfire** [comment]
Do we know when o1-preview will go GA for Copilot and if there will be any limits? I've been trialling Cursor and it's light years ahead in terms of models and AI features. Copilot is almost unusable for me after experiencing Claude 3.5 Sonnet in Cursor. I'm hoping o1 bridges the gap.

**Reply by ohofherr** [reply]
Seems like GitHub Copilot is no longer no1

**Comment by cmp-nct** [comment]
For anything that is not extremely trivial, only o1-preview has given me good responses.
The other models are, frankly, trash for development.
However, they are very good for repetitive auto-completion.

**Reply by Anselmoo** [reply]
@cmp-nct, can you please provide a few more details about the repetitive auto-completion? I'm curious to know more.

<img width="577" alt="image" src="https://github.com/user-attachments/assets/f3cdb1ba-546a-4e2d-a002-e888ef60b779">

For me, ` o1-preview` is a big help, especially for refactoring, because it explains the idea behind it in detail. One thing that I recently observed is that sometimes, the files are not accurately separated.

<img width="562" alt="image" src="https://github.com/user-attachments/assets/73ef52dd-a849-452f-815f-332d4314e968">

Everything goes into one file ...

**Comment by clover-yan** [comment]
Do we have a rate limit for o1 series models?

**Comment by zseirafi** [comment]
Hi, we signed up codewrx enterprise for the waitlist to fine tune models. Would love to get access soon. thanks

**Comment by staytuned-gh** [comment]
@2percentsilk We are facing a limit while using o1 model, any solution? And this is happening everyday...Please help

![image](https://github.com/user-attachments/assets/218aced5-f014-492c-bd8e-e03a859854cc)

**Comment by tp-midud** [comment]
Can we have this for `neovim`/`vim` as well?

**Reply by lovecigare** [reply]
I think "yes"

**Comment by Epicism** [comment]
O1 keeps overwriting my code when I ask it to document it. It will replace a function with 
```
/// Comment
/// Comment
fn test() {
     ... Code here
}
```

Not sure if anyone else is experiencing this.

**Comment by lovecigare** [comment]
Can i also try?

**Comment by nikzan** [comment]
Do the O1 models appear in JetBrains IDEs or is this feature only for VS Code for this moment?

**Comment by LIHUA919** [comment]
agree












刘立华
***@***.***






---Original---
From: "Никита ***@***.***&gt;
Date: Tue, Dec 10, 2024 18:12 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] OpenAI o1 on GitHub (Limited PublicPreview) Feedback Thread (Discussion #139125)




 
Do the O1 models appear in JetBrains IDEs or is this feature only for VS Code for this moment?
